test/unit/stats.js

```diff
@@ -0,0 +1,311 @@
+      assert.strictEqual(49628525, stats.channels[2]['sum']);
+      assert.strictEqual(6450556071, stats.channels[2]['squaresSum']);
+      assert.strictEqual(116.38959896810506, stats.channels[2]['mean']);
+      assert.strictEqual(39.7669551046809, stats.channels[2]['stdev']);
+
+      done();
+    });
+  });
+
+  it('Grayscale GIF with alpha', function (done) {
+    sharp(fixtures.inputGifGreyPlusAlpha).stats(function (err, stats) {
+      if (err) throw err;
+      assert.strictEqual(false, stats.isOpaque);
+
+      // gray channel
+      assert.strictEqual(0, stats.channels[0]['min']);
+      assert.strictEqual(101, stats.channels[0]['max']);
+      assert.strictEqual(101, stats.channels[0]['sum']);
+      assert.strictEqual(10201, stats.channels[0]['squaresSum']);
+      assert.strictEqual(50.5, stats.channels[0]['mean']);
+      assert.strictEqual(71.4177848998413, stats.channels[0]['stdev']);
+
+      // alpha channel
+      assert.strictEqual(0, stats.channels[1]['min']);
+      assert.strictEqual(255, stats.channels[1]['max']);
+      assert.strictEqual(255, stats.channels[1]['sum']);
+      assert.strictEqual(65025, stats.channels[1]['squaresSum']);
+      assert.strictEqual(127.5, stats.channels[1]['mean']);
+      assert.strictEqual(180.31222920256963, stats.channels[1]['stdev']);
+
+      done();
+    });
+  });
+
+  it('Stream in, Callback out', function (done) {
+    const readable = fs.createReadStream(fixtures.inputJpg);
+    const pipeline = sharp().stats(function (err, stats) {
+      if (err) throw err;
+      assert.strictEqual(true, stats.isOpaque);
+
+      // red channel
+      assert.strictEqual(0, stats.channels[0]['min']);
+      assert.strictEqual(255, stats.channels[0]['max']);
+      assert.strictEqual(615101275, stats.channels[0]['sum']);
+      assert.strictEqual(83061892917, stats.channels[0]['squaresSum']);
+      assert.strictEqual(101.44954540768993, stats.channels[0]['mean']);
+      assert.strictEqual(58.373870588815414, stats.channels[0]['stdev']);
+
+      // green channel
+      assert.strictEqual(0, stats.channels[1]['min']);
+      assert.strictEqual(255, stats.channels[1]['max']);
+      assert.strictEqual(462824115, stats.channels[1]['sum']);
+      assert.strictEqual(47083677255, stats.channels[1]['squaresSum']);
+      assert.strictEqual(76.33425255128337, stats.channels[1]['mean']);
+      assert.strictEqual(44.03023262954866, stats.channels[1]['stdev']);
+
+      // blue channel
+      assert.strictEqual(0, stats.channels[2]['min']);
+      assert.strictEqual(255, stats.channels[2]['max']);
+      assert.strictEqual(372986756, stats.channels[2]['sum']);
+      assert.strictEqual(32151543524, stats.channels[2]['squaresSum']);
+      assert.strictEqual(61.51724663436759, stats.channels[2]['mean']);
+      assert.strictEqual(38.96702865090125, stats.channels[2]['stdev']);
+
+      done();
+    });
+    readable.pipe(pipeline);
+  });
+
+  it('Stream in, Promise out', function (done) {
```

Comment:
This test should be able to return the Promise instance, which will then remove the need to use `done` and `catch`.

Reply:
`.stats()` does return a promise in this test. I am not sure if I got your point. A similar test exists for `metadata()` as well.

Reply:
mocha deals with tests that return (and reject) a `Promise`, e.g.:
```javascript
  it('Stream in, Promise out', function () {
    const pipeline = sharp();

    fs
      .createReadStream(fixtures.inputJpg)
      .pipe(pipeline);

    return pipeline
      .stats()
      .then(function (stats) {
        ...
      });
  });
```